File: app/src/main/java/com/ywsh/widget/CircleView.java
package com.ywsh.widget;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.View;

public class CircleView extends View {

    private static final float RADIUS = 100.0f;

    private Point point;

    private Paint mPaint;

    public CircleView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        mPaint.setColor(Color.BLUE);
    }

    @Override
    protected void onDraw(Canvas canvas){
        if (point == null){
            point = new Point(RADIUS, RADIUS);
            drawCircle(canvas);
        } else {
            drawCircle(canvas);
        }
        invalidate();
    }

    private void drawCircle(Canvas canvas) {
        float x = point.getX();
        float y = point.getY();
        canvas.drawCircle(x, y, RADIUS, mPaint);
    }

    public Point getPoint(){
        return point;
    }

    public void setPoint(Point point){
        this.point = point;
    }
}

File: app/src/main/java/com/ywsh/widget/CircleView.kt
package com.ywsh.widget

import android.animation.AnimatorSet
import android.animation.ObjectAnimator
import android.animation.ValueAnimator
import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View

class CircleView(context: Context?, attrs: AttributeSet?) : View(context, attrs) {
    var point: Point? = null
    var color: String? = null
    private val mPaint: Paint = Paint(Paint.ANTI_ALIAS_FLAG)
    override fun onDraw(canvas: Canvas) {
        if (color != null) {
            mPaint.color = Color.parseColor(color)
        }
        if (point == null) {
            point = Point(RADIUS, RADIUS)
            drawCircle(canvas)
        } else {
            drawCircle(canvas)
        }
        invalidate()
    }

    private fun drawCircle(canvas: Canvas) {
        val x = point!!.x
        val y = point!!.y
        canvas.drawCircle(x, y, RADIUS, mPaint)
    }

    var animSet: AnimatorSet? = null

    fun startAnimation() {
        //let run apply also
        animSet?.start() ?: let {

            val anim = ObjectAnimator.ofObject(this, "point", PointEvaluator(),
                    Point(100.0f, 100.0f), Point(500.0f, 500.0f)).apply {
                repeatCount = ValueAnimator.INFINITE
                repeatMode = ValueAnimator.REVERSE
            }
            val anim2 = ObjectAnimator.ofObject(this, "color", ColorEvaluator(),
                    "#0000FF", "#FF0000").apply {
                repeatCount = ValueAnimator.INFINITE
                repeatMode = ValueAnimator.REVERSE
            }
            animSet = AnimatorSet()
            animSet?.apply {
                play(anim)?.with(anim2)
                duration = 2000
                start()
            }
        }
    }

    companion object {
        private const val RADIUS = 100.0f
    }

    init {
        mPaint.color = Color.BLUE
    }
}